File: xlockmore/5.x/5.65/xlockmore-5.65/etc/xlocksat.pl
#!/usr/bin/perl -T -w
# xlocksat
#require "ctime.pl";
use POSIX qw( strftime );

#local($hour) = `/bin/date "+%H:"`;
#local($hour) = (localtime)[2];
local($hour) = strftime('%H', localtime() );
#printf("%d\n", $hour);
local($saturation) = (12 - abs($hour - 12)) / 12 ;
printf("%.2f\n", $saturation);

File: xlockmore/5.x/5.51/xlockmore-5.51/etc/xlocksat.py
#!/usr/bin/env python
# xlocksat

import datetime
import sys
def put(string):
  sys.stdout.write(string)

t = datetime.datetime.now().time()
#print t.hour
saturation = round((12.0 - abs(t.hour - 12.0)) / 12.0, 2)
put(str(saturation))
